src/TMap.cpp

```diff
@@ -2217,8 +2229,17 @@ void TMap::deleteMapLabel(int area, int labelId)
         return;
     }
 
-    if (pA->mMapLabels.remove(labelId)) {
-        mUnsavedMap = true;
+    auto label = pA->mMapLabels.take(labelId);
+    if (!label.pos.isNull() || !label.text.isEmpty() || label.bgColor != QColorConstants::Black || label.fgColor != QColorConstants::Black) {
+        // If any of the above tests are false then we can take it that we do
+        // not have a "default constructed" label - i.e. a real one and not
+        // one that the QMap<T1, T2>::take(const T1&) has created for us in the
+        // absence of an actual TMapLabel.
+        // The TMapLabel default constructor sets the temporary member to false
+        // so we can safely rely on it being true for a permanent one:
```

Comment:
:blush: Oops, got that the wrong way around! Need to change that to "... true for a temporary one:"

Reply:
Fixed by: https://github.com/Mudlet/Mudlet/pull/6285/commits/18e64d1b843c0144fbd752f47a8215e3954c9ce4